src/components/Header.js

```diff
@@ -1,7 +1,17 @@
 import React from 'react';
 import { useState } from "react";
+import { product_list } from "./product-list";
+
+function filterData(searchInput,productList) {
+    const filterData = productList.filter((product) =>
+    product.pname.includes(searchInput)
```

Comment:
product?.pname.includes(searchInput)
 
 Add a question mark which will check if product is defined. And if not defined, it will not try to access pname. This will protect the code from giving error when the pname is not present.